### PR TITLE
Clean up exceptions and `badRequest` results in `QuestionController`

### PR DESCRIPTION
**Pull request by natsid**
### Description
* Refactor some `try`/`catch` blocks
* Make bad request messages more consistent for the same exceptions

### CODE REVIEW

**Review by hmhsu — APPROVED** [review]
Thanks for doing this!

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java [hunk 1]
```diff
@@ -90,41 +90,43 @@ public QuestionController(
...
-      ImmutableList<RepeaterQuestionDefinition> repeaterQuestionDefinitions =
-          service
-              .getReadOnlyQuestionService()
-              .toCompletableFuture()
-              .join()
-              .getUpToDateRepeaterQuestions();
       return ok(editView.renderNewQuestionForm(request, questionType, repeaterQuestionDefinitions));
     } catch (UnsupportedQuestionTypeException e) {
       return badRequest(e.getMessage());
-    } catch (IllegalArgumentException e) {
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:112]: question: what is throwing the `IllegalArgumentException`? Is it something in the view that still needs to be caught?

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java [hunk 2]
```diff
@@ -201,9 +203,7 @@ public Result update(Request request, Long id, String questionType) {
               .setLifecycleStage(LifecycleStage.DRAFT)
               .build();
     } catch (UnsupportedQuestionTypeException e) {
-      throw new RuntimeException(
-          "Failed while trying to update a question that was already created for question type "
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:205]: suggestion (non-blocking): Can we move this message into a comment, so we still have info for when this case happens?

**Review by carolinedanzi — APPROVED** [review]
Thanks for doing this cleanup!

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java (hunk 2 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:205]: Good idea

File: universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java (hunk 1 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java:112]: Good question. I changed a line above from `QuestionType.valueOf` (which can throw an `IllegalArgumentException`), to using `QuestionType.of`, which is a method on `QuestionType` that does some nicer handling and potentially throws an `InvalidQuestionTypeException`. So this should be fine.

**Pull request merged**